services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/router/AuthorizationRouterImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.softgroup.authorization.impl.router;
+
+import com.softgroup.authorization.api.router.AuthorizationRequestHandler;
+import com.softgroup.authorization.api.router.AuthorizationRouterHandler;
+import com.softgroup.common.protocol.Request;
+import com.softgroup.common.protocol.Response;
+import com.softgroup.common.router.api.AbstractRouterHandler;
+import com.softgroup.common.router.api.Handler;
+import com.softgroup.common.router.api.factories.CommandHandlerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by Виктор on 03.03.2017.
+ */
+@Component
+public class AuthorizationRouterImpl extends AbstractRouterHandler<AuthorizationRequestHandler>
+                                     implements AuthorizationRouterHandler{
+
+    @Autowired
+    private CommandHandlerFactory<AuthorizationRequestHandler> handlerFactory;
```

Comment:
Now I see two variants, 1. Create abstract class AbstractHandlerFactory and inject it into router. 2. Create interface HandlerFactory and inject into router.  Which of these variants would be better?